File: android/src/main/java/ee/forgr/capacitor/apptrackingtransparency/AppTrackingTransparencyPlugin.java
package ee.forgr.capacitor.apptrackingtransparency;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

/**
 * App Tracking Transparency plugin for Android.
 *
 * Since App Tracking Transparency is an iOS-only framework,
 * this Android implementation always returns 'authorized' status
 * to allow transparent cross-platform usage.
 */
@CapacitorPlugin(name = "AppTrackingTransparency")
public class AppTrackingTransparencyPlugin extends Plugin {

    private final String pluginVersion = "8.1.0";

    /**
     * Returns the current tracking authorization status.
     * On Android, this always returns 'authorized' since ATT is iOS-only.
     */
    @PluginMethod
    public void getStatus(PluginCall call) {
        JSObject ret = new JSObject();
        // Android doesn't have App Tracking Transparency
        // Return 'authorized' for transparent cross-platform usage
        ret.put("status", "authorized");
        call.resolve(ret);
    }

    /**
     * Requests tracking authorization from the user.
     * On Android, this always returns 'authorized' since ATT is iOS-only.
     */
    @PluginMethod
    public void requestPermission(PluginCall call) {
        JSObject ret = new JSObject();
        // Android doesn't have App Tracking Transparency
        // Return 'authorized' for transparent cross-platform usage
        ret.put("status", "authorized");
        call.resolve(ret);
    }

    /**
     * Returns the plugin version.
     */
    @PluginMethod
    public void getPluginVersion(final PluginCall call) {
        try {
            final JSObject ret = new JSObject();
            ret.put("version", this.pluginVersion);
            call.resolve(ret);
        } catch (final Exception e) {
            call.reject("Could not get plugin version", e);
        }
    }
}

File: ios/Sources/AppTrackingTransparencyPlugin/AppTrackingTransparencyPlugin.swift
import Foundation
import Capacitor
import AppTrackingTransparency

@objc(AppTrackingTransparencyPlugin)
public class AppTrackingTransparencyPlugin: CAPPlugin, CAPBridgedPlugin {
    private let pluginVersion: String = "8.1.4"
    public let identifier = "AppTrackingTransparencyPlugin"
    public let jsName = "AppTrackingTransparency"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "getStatus", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "requestPermission", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "getPluginVersion", returnType: CAPPluginReturnPromise)
    ]

    @objc func getStatus(_ call: CAPPluginCall) {
        if #available(iOS 14, *) {
            let status = ATTrackingManager.trackingAuthorizationStatus
            call.resolve(["status": statusToString(status)])
        } else {
            // iOS < 14 doesn't require tracking permission
            call.resolve(["status": "authorized"])
        }
    }

    @objc func requestPermission(_ call: CAPPluginCall) {
        if #available(iOS 14, *) {
            ATTrackingManager.requestTrackingAuthorization { status in
                DispatchQueue.main.async {
                    call.resolve(["status": self.statusToString(status)])
                }
            }
        } else {
            // iOS < 14 doesn't require tracking permission
            call.resolve(["status": "authorized"])
        }
    }

    @objc func getPluginVersion(_ call: CAPPluginCall) {
        call.resolve(["version": self.pluginVersion])
    }

    @available(iOS 14, *)
    private func statusToString(_ status: ATTrackingManager.AuthorizationStatus) -> String {
        switch status {
        case .authorized:
            return "authorized"
        case .denied:
            return "denied"
        case .notDetermined:
            return "notDetermined"
        case .restricted:
            return "restricted"
        @unknown default:
            return "notDetermined"
        }
    }
}
